roles/matrix-synapse/tasks/synapse/setup_install.yml

```diff
@@ -119,3 +128,21 @@
     src: "{{ role_path }}/templates/synapse/usr-local-bin/matrix-synapse-register-user.j2"
     dest: "{{ matrix_local_bin_path }}/matrix-synapse-register-user"
     mode: 0755
+
+- set_fact:
+    matrix_synapse_modules: |
+      {{
+        matrix_synapse_modules|default([])
+        +
+        [
+          {
+            "module": "s3_storage_provider.S3StorageProviderBackend",
+            "config":{
+              "store_local": {{ matrix_synapse_s3_media_store_on_s3 }},
+              "store_remote": {{ matrix_synapse_s3_media_store_on_local_filesystem }},
+              "store_synchronous": {{ matrix_synapse_media_storage_provider_s3_store_synchronous }},
+              "config": {{ matrix_synapse_media_storage_provider_s3_config }}
+            }
+          }
+        ]
+      }}
```

Comment:
Here I injected the whole module config. But it does not get in the homeserver.yaml

Reply:
You should inject into a new variable (`matrix_synapse_media_storage_providers: []` in `defaults/main.yml`).

Then you should have this in `homeserver.yaml.j2`:

```yaml
media_storage_providers: {{ matrix_synapse_media_storage_providers|to_json }}
```